sdk/keyvault/cleanup/cleanup.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief This sample provides the code implementation to use the Key Vault Secrets SDK client for
+ * C++ to create, get, update, delete and purge a secret.
+ *
+ * @remark The following environment variables must be set before running the sample.
+ * - AZURE_KEYVAULT_URL:  To the Key Vault account URL.
+ * - AZURE_TENANT_ID:     Tenant ID for the Azure account.
+ * - AZURE_CLIENT_ID:     The Client ID to authenticate the request.
+ * - AZURE_CLIENT_SECRET: The client secret.
+ *
+ */
+
+#include "get_env.hpp"
+
+#include <azure/identity.hpp>
+#include <azure/keyvault/certificates.hpp>
+#include <azure/keyvault/keys.hpp>
+#include <azure/keyvault/secrets.hpp>
+#include <chrono>
+#include <iostream>
+
+using namespace Azure::Security::KeyVault::Secrets;
+using namespace Azure::Security::KeyVault::Certificates;
+using namespace Azure::Security::KeyVault::Keys;
+using namespace std::chrono_literals;
+
+int main()
+{
+  auto tenantId = std::getenv("AZURE_TENANT_ID");
+  auto clientId = std::getenv("AZURE_CLIENT_ID");
+  auto clientSecret = std::getenv("AZURE_CLIENT_SECRET");
+  auto credential
+      = std::make_shared<Azure::Identity::ClientSecretCredential>(tenantId, clientId, clientSecret);
+
+  SecretClient secretClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
+  KeyClient keyClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
+  CertificateClient certClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
+
+  try
+  {
+    std::vector<DeleteCertificateOperation> certOps;
```

Comment:
Silly question (and yes, I understand that this is existing code): Why does the StartXxx APIs return a specialized Operation object instead of returning a std::unique_ptr<Operation>. The current API shape prevents you from aggregating the Operation objects returned from the various LRO returning APIs, which is somewhat awkward.

Reply:
That was the architecture decision , i just folowed it